src/data/types.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * The search result object
+ * @typedef {Object} MediaResult
+ * @property {number} result_count
+ * @property {number} page_count
+ * @property {number} page_size
+ * @property {MediaResult[]} results
```

Comment:
It looks like MediaResult is also generic here: https://github.com/WordPress/openverse-frontend/pull/173/files#diff-a6daedb006e158dcd94eb06a42f670f76b3c8ffe49b9dfdee53a53f6c492c0a9R51

Should it be this?

```suggestion
 * The search result object
 * @template {MediaResult} T
 * @typedef {Object} MediaResult
 * @property {number} result_count
 * @property {number} page_count
 * @property {number} page_size
 * @property {T[]} results
```